CMakeLists.txt

```diff
@@ -195,11 +198,16 @@ FetchContent_Declare(
 ################################
 # ABSEIL
 ################################
-set(BUILD_TESTING OFF CACHE BOOL "Don't build tests for abseil" FORCE)
-# For CMake 3.8 and up this will be the default in a future version of abseil
+
+set(ABSL_ENABLE_INSTALL ON)
+set(ABSL_USE_SYSTEM_INCLUDES ON)
 set(ABSL_PROPAGATE_CXX_STD ON)
-add_subdirectory(third_party/abseil-cpp  EXCLUDE_FROM_ALL)
-include_directories(SYSTEM third_party/abseil-cpp/)
+FetchContent_Declare(
+        abseil
+        GIT_REPOSITORY https://github.com/abseil/abseil-cpp.git
+        GIT_TAG 258e5a15759cc3d122d4a4826bc499af91d40aa9 # 2023-12-21
+)
+FetchContent_MakeAvailable(abseil)
```

Comment:
Calling `MakeAvailable` individually is an anti-pattern. You're supposed to first declare all the dependencies and then make them available all at once. See https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable form more information